flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/batch/connectors/hive/HiveTableInputFormat.java

```diff
@@ -0,0 +1,275 @@
+			RowTypeInfo rowTypeInfo) {
+		super(jobConf.getCredentials());
+		this.rowTypeInfo = checkNotNull(rowTypeInfo, "rowTypeInfo can not be null.");
+		this.jobConf = new JobConf(jobConf);
+		this.isPartitioned = checkNotNull(isPartitioned, "isPartitioned can not be null.");
+		this.partitionColNames = partitionColNames;
+		this.partitions = checkNotNull(partitions, "partitions can not be null.");
+	}
+
+	@Override
+	public void open(HiveTableInputSplit split) throws IOException {
+		this.hiveTablePartition = split.getHiveTablePartition();
+		StorageDescriptor sd = hiveTablePartition.getStorageDescriptor();
+		jobConf.set(INPUT_DIR, sd.getLocation());
+		try {
+			this.mapredInputFormat = (InputFormat)
+				Class.forName(sd.getInputFormat(), true, Thread.currentThread().getContextClassLoader()).newInstance();
+		} catch (Exception e) {
+			throw new FlinkHiveException("Unable to instantiate the hadoop input format", e);
+		}
+		ReflectionUtils.setConf(mapredInputFormat, jobConf);
+		if (this.mapredInputFormat instanceof Configurable) {
+			((Configurable) this.mapredInputFormat).setConf(this.jobConf);
+		} else if (this.mapredInputFormat instanceof JobConfigurable) {
+			((JobConfigurable) this.mapredInputFormat).configure(this.jobConf);
+		}
+		this.recordReader = this.mapredInputFormat.getRecordReader(split.getHadoopInputSplit(),
+			jobConf, new HadoopDummyReporter());
+		if (this.recordReader instanceof Configurable) {
+			((Configurable) this.recordReader).setConf(jobConf);
+		}
+		key = this.recordReader.createKey();
+		value = this.recordReader.createValue();
+		this.fetched = false;
+		try {
+			deserializer = (Deserializer) Class.forName(sd.getSerdeInfo().getSerializationLib()).newInstance();
+			Configuration conf = new Configuration();
+			//properties are used to initialize hive Deserializer properly.
+			Properties properties = HiveTableUtil.createPropertiesFromStorageDescriptor(sd);
+			SerDeUtils.initializeSerDe(deserializer, conf, properties, null);
+			structObjectInspector = (StructObjectInspector) deserializer.getObjectInspector();
+			structFields = structObjectInspector.getAllStructFieldRefs();
+		} catch (Exception e) {
+			throw new FlinkHiveException("Error happens when deserialize from storage file.", e);
+		}
+	}
+
+	@Override
+	public HiveTableInputSplit[] createInputSplits(int minNumSplits)
+			throws IOException {
+		List<HiveTableInputSplit> hiveSplits = new ArrayList<>();
+		int splitNum = 0;
+		for (HiveTablePartition partition : partitions) {
+			StorageDescriptor sd = partition.getStorageDescriptor();
+			InputFormat format;
+			try {
+				format = (InputFormat)
+					Class.forName(sd.getInputFormat(), true, Thread.currentThread().getContextClassLoader()).newInstance();
+			} catch (Exception e) {
+				throw new FlinkHiveException("Unable to instantiate the hadoop input format", e);
+			}
+			ReflectionUtils.setConf(format, jobConf);
+			jobConf.set(INPUT_DIR, sd.getLocation());
```

Comment:
Don't we need to clone the JobConf here, since the input dir can be different for each partition?

Reply:
`createInputSplits` method is only called in JM to calc splits, so there no need to clone the JobConf.

Reply:
I see. Thanks.